src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -36,21 +37,27 @@ public sealed class TensorFlowTransform : ITransformer, ICanSaveModel
     {
         public sealed class Arguments : TransformInputBase
         {
+            [Argument(ArgumentType.Required, HelpText = "TensorFlow model used by the transform. Please see https://www.tensorflow.org/mobile/prepare_models for more details.", SortOrder = 0)]
+            public string Model;
 
-            [Argument(ArgumentType.Required, HelpText = "This is the frozen protobuf model file. Please see https://www.tensorflow.org/mobile/prepare_models for more details.", ShortName = "model", SortOrder = 0)]
-            public string ModelFile;
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Indicator for frozen models", ShortName = "frozen", SortOrder = 1)]
+            public bool IsFrozen = TensorFlowEstimator.Defaults.IsFrozen;
 
-            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "The names of the model inputs", ShortName = "inputs", SortOrder = 1)]
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "The names of the model inputs", ShortName = "inputs", SortOrder = 2)]
             public string[] InputColumns;
 
-            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "The name of the outputs", ShortName = "outputs", SortOrder = 2)]
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "The name of the outputs", ShortName = "outputs", SortOrder = 3)]
             public string[] OutputColumns;
         }
 
         private readonly IHost _host;
+        private static string _savedModel;
         private const string RegistrationName = "TensorFlowTransform";
+        private const string TempSavedModelDirName = "MLNET_TensorFlowTransform_SavedModel";
+        private const string TempSavedModelZipName = "SavedModel.zip";
```

Comment:
>private const string TempSavedModelZipName = "SavedModel.zip"; [](start = 8, length = 62)

What will happen I am working on two different pipelines with TensorflowTransforms executing in parallel? I think having same name for .zip (`SavedModel.zip`) and model_dir (`MLNET_TensorFlowTransform_SavedModel`)will create a problem.

Let see what @Eric suggest. However, when I did initially, my approach was as follows
- In the constructor, load all the files in un-frozen model in to dictionary (model dictionary) where keys are the file name and value is the stream of file.
- Load the model from the location that user provided.
- When saving, save the streams in the model dictionary into ML.Net stream.
- When loading from ML.Net stream, load the streams into the model dictionary, create temporary folder and write all the stream into the folder. Load the model into TF using that temporary folder. Delete the model once loading is done.

Reply:
This obviously have a limitation on the size of model.

---
In reply to: [217133381](https://github.com/dotnet/machinelearning/pull/853#discussion_r217133381) [](ancestors = 217133381)

Reply:
True. I have modified the logic so we do the load&restore in-memory (using a dictionary).  

---
In reply to: [217134254](https://github.com/dotnet/machinelearning/pull/853#discussion_r217134254) [](ancestors = 217134254,217133381)